src/persist-client/src/internal/compact.rs

```diff
@@ -254,14 +275,233 @@ impl Compactor {
         // Compaction is cpu intensive, so be polite and spawn it on the
         // CPU heavy runtime.
         let compact_span = debug_span!("compact::consolidate");
-        cpu_heavy_runtime
+        parts_cpu_heavy_runtime
             .spawn_named(
                 || "persist::compact::consolidate",
                 compact_blocking.instrument(compact_span),
             )
             .await?
     }
 
+    /// WIP: grab the first run from each batch, then the second, etc.
+    fn order_runs<T, D>(req: &CompactReq<T>) -> Vec<(&HollowBatch<T>, &[PartialBatchKey])>
+    where
+        T: Timestamp + Lattice + Codec64,
+        D: Semigroup + Codec64 + Send + Sync,
+    {
+        let mut all_runs = vec![];
+        for batch in &req.inputs {
+            all_runs.push((batch, batch.runs()));
+        }
+
+        // map our (Batch, [Runs]) to [(Batch, Run), ...] cycling through the input batches
+        let mut finished_iterators = 0;
+        let mut ordered_runs = vec![];
+        loop {
+            for (batch, runs) in &mut all_runs {
+                match runs.next() {
+                    Some(run) => ordered_runs.push((*batch, run)),
+                    None => finished_iterators += 1,
+                }
+            }
+
+            if finished_iterators >= all_runs.len() {
+                break;
+            }
+        }
+
+        ordered_runs
+    }
+
+    // WIP: Compacts runs together, pulling down one part at a time from each. If the runs are
+    //      already sorted, the result should be a single run. Memory is bounded by the # of
+    //      runs passed into this function, consuming at most (runs + 1) * blob_target_size
+    async fn compact_runs<'a, T, D>(
```

Comment:
I think you should be able to get rid of the `'a` param and just have all the inputs be `&`?

Reply:
ugh, one would think. was getting the weirdest rustc errors that led me to an open issue where sometimes lifetimes aren't inferred correctly on async functions

Reply:
ah gotcha. comment then plz